client/src/main/java/io/pravega/client/segment/impl/EventSegmentReaderImpl.java

```diff
@@ -28,6 +29,10 @@
 @ToString
 class EventSegmentReaderImpl implements EventSegmentReader {
 
+    /* Partial data timeout. This timeout is the maximum amount of time the reader will wait in the case of
+     * partial data being received by the client. After this timeout the client will resend the request. */
+    static final long READ_TIMEOUT_MS = 500;
```

Comment:
This seems really aggressive. I don't see any reason this should be more aggressive than or different from `RevisionedStreamClientImpl.READ_TIMEOUT_MS`. In both cases we know the server has data and we are waiting to receive it.

Reply:
ok.Changed it to 30 seconds.